src/gui/Measurement.cc

```diff
@@ -128,17 +138,36 @@ QString Measurement::statemachine(QPoint mouse)
         obj3=qglview->selected_obj[2];
         if(obj1.type == SelectionType::SELECTION_POINT && obj2.type == SelectionType::SELECTION_POINT && obj3.type == SelectionType::SELECTION_POINT)
         {
-          Vector3d side1=(obj2.p1-obj1.p1).normalized();
-          Vector3d side2=(obj2.p1-obj3.p1).normalized();
-          ang=acos(side1.dot(side2))*180.0/3.14159265359;
+	  p1=obj1.p1;
+          p2=obj2.p1;
+	  p3=obj2.p1;
+	  p4=obj3.p1;
+	  goto display_angle;
```

Comment:
This makes code pretty hard to read - try to avoid goto and split out the code into a function or lambda instead. In general, large switch-case statements are hard to follow.